docs/sources/developer/architecture/components/api.md

```diff
@@ -0,0 +1,78 @@
+# API
+
+The Faro API is the primary interface for end-users to interact with Faro. It is the central place where signals are
+sent to but it also contains helpers methods for other components like metas.
+
+## Signals
+
+### Events
+
+Events are signals related to tracking various actions like mouse clicks as well as other events like transitioning to
+another page.
+
+Methods:
+
+- `pushEvent` - sends an event
+
+### Exceptions
```

Comment:
See below - question around whether ###Errors or ###Error Reporting would be the right terminology (or not)

Reply:
TBD later